File: sentimental-mario-less/mario.py
from cs50 import get_int

# get height
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break
# iterate over the given height [i = 0, 1, 2, ... , (height - 1)]
for i in range(height):
    # print appriopirate blank spaces without \n in the end
    print(" " * (height - i - 1), end="")
    # print appriopirate hashes after blank spaces
    print("#" * (i + 1))

File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //Reminding the complier that the functions exists
    int getHeight(void);
    void printPyramid(int);

    //Getting the height of the pyramid
    int height = getHeight();

    //Printing the pyramid
    printPyramid(height);

}

//function for getting height
int getHeight(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}

//function for printing pyramid
void printPyramid(int height)
{
    //main loop that iterates until i equals height given
    for (int i = 0; i < height; i++)
    {
        //loop for printing height - 1 number of blank spaces
        for (int k = height - 1; k > i; k--)
        {
            printf(" ");
        }

        //loop for printing n hashes per nth iteration after the previous loop puts height - 1 number of blank spaces
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n"); //new line each time the two previous loops conclude
    }
}